plugins/services/src/js/stores/SDKEndpointStore.js

```diff
@@ -0,0 +1,149 @@
+
+import SDKEndpointActions from "../events/SDKEndpointActions";
+import SDKServiceEndpoint from "../structs/SDKServiceEndpoint";
+
+class SDKEndpointStore extends GetSetBaseStore {
+  constructor() {
+    super(...arguments);
+
+    this.getSet_data = {
+      services: {}
+    };
+
+    PluginSDK.addStoreConfig({
+      store: this,
+      storeID: this.storeID,
+      listenAlways: true
+    });
+
+    this.dispatcherIndex = AppDispatcher.register(payload => {
+      if (payload.source !== SERVER_ACTION) {
+        return false;
+      }
+
+      const { type, data } = payload.action;
+      switch (type) {
+        case REQUEST_SDK_ENDPOINTS_SUCCESS:
+          this.processNewEndpoints(data.serviceId, data.endpoints);
+          break;
+        case REQUEST_SDK_ENDPOINTS_ERROR:
+          this.setService(data.serviceId, {
+            endpoints: [],
+            totalLoadingEndpointsCount: -1,
+            error: data.error.description
+          });
+          break;
+        case REQUEST_SDK_ENDPOINT_SUCCESS:
+          this.processNewEndpoint(
+            data.serviceId,
+            data.endpointName,
+            data.endpointData,
+            data.contentType
+          );
+          break;
+        case REQUEST_SDK_ENDPOINT_ERROR:
+          this.setService(data.serviceId, {
+            endpoints: [],
+            totalLoadingEndpointsCount: -1,
+            error: data.error
+          });
+          break;
+      }
+
+      return true;
+    });
+  }
+
+  getSDKEndpointService(serviceId) {
+    const service = this.get("services")[serviceId];
+
+    if (!service || !service.endpoints) {
+      return null;
+    }
+
+    const sdkEndpointList = Object.keys(service.endpoints).map(endpointName => {
+      const endpoint = service.endpoints[endpointName];
+
+      return new SDKServiceEndpoint({
+        endpointName,
+        endpointData: endpoint.endpointData,
+        contentType: endpoint.contentType
+      });
+    });
+
+    return Object.assign({}, service, {
+      endpoints: sdkEndpointList
+    });
+  }
+
+  setService(serviceId, serviceData) {
+    const services = Object.assign({}, this.get("services"), {
+      [serviceId]: {
+        endpoints: serviceData.endpoints,
+        totalLoadingEndpointsCount: serviceData.totalLoadingEndpointsCount,
+        error: serviceData.error
+      }
+    });
+    this.set({ services });
+  }
+
+  processNewEndpoints(serviceId, endpointsArr) {
+    const endpoints = endpointsArr.reduce(function(acc, curr) {
+      acc[curr] = {};
+
+      return acc;
+    }, {});
+
+    this.setService(serviceId, {
+      endpoints,
+      totalLoadingEndpointsCount: endpointsArr.length,
+      error: ""
+    });
+
+    endpointsArr.forEach(function(endpoint) {
+      SDKEndpointActions.fetchEndpoint(serviceId, endpoint);
+    }, this);
+  }
+
+  processNewEndpoint(serviceId, endpointName, endpointData, contentType) {
+    const service = this.get("services")[serviceId];
```

Comment:
We do have a service setter let's add a service getter that provides a copy so that we don't need to add all of the `Object.assign` calls.

Reply:
Good idea, created a getServices method